doc/topics/core-structure.md

```diff
@@ -0,0 +1,202 @@
+- C++ "native" Source (`src/`)
+- Tests & Benchmarks (`test/` & `benchmark/`)
+- Internal / Build Tooling (`tools/`)
+- Dependencies (`deps/`)
+- .dotfiles
+
+Below is a summary of the parts, ordered as per the list above.
+
+
+### CAPITALS files
+
+These are pretty important, and describe things such as our [GOVERNANCE][], [CONTRIBUTING][], and [CODE_OF_CONDUCT][] policies.
+The [LICENSE][] is also part of these.
+
+If you plan on building node, it is recommended you check out the [BUILDING][]
+document.
+
+Typically these are only found at the root directory of the project.
+
+
+### Documentation (`doc/`)
+
+All the bundled documentation assets, including the [API docs][] that are
+available on the Nodejs.org website, can be found within the `doc/` folder.
+
+The API docs are constructed from the markdown files within the `doc/api/`
+folder.
+
+The documentation also includes topic documents such as this one such as this,
+how-to guides, the `node.1` [manual page][], and an archive of Technical
+Committee meeting notes.
+
+
+### Top-level Build Tooling
+
+If you are planning on making modifications to Node.js, you will need to be able
+to build the `node` binary.
+Full instructions can be found in the [BUILDING][] document.
+
+The top-level (root directory) build files help you achieve this, most notably
+[`configure`][] & [`Makefile`][] for UNIX systems and [`vcbuild.bat`][] for
+Windows.
+
+There are also some other build files in the root directory, such as the
+`node.gyp` gyp build configuration. Though in most cases, those will not need to
+be edited.
+
+
+### JavaScript Library (`lib/`)
+
+The JavaScript library found within the `lib/` folder is one of the most
+important parts of Node.js, and you will most likely be interacting with the
+files within if you are making code modifications.
+
+All of node's exposed API can be found exported from these files.
+Specifically, any `.js` file directly within the `lib/` folder is exported as a
+module that can be `require()`'d in user code.
+Node.js also has some internal files, for utilities not intended to be exposed.
+Those can be found in the `lib/internal/` folder, and files within are not
+accessible to JavaScript code outside of the node core.
+
+Additionally, Node.js's own JavaScript entry point can be found at
+`lib/internal/bootstrap_node.js`. This file determines how to execute the user
+code, and also helps set up the `process` global object, importing lots from
+`lib/internal/process/`.
+
+As you inspect these files, you will occasionally see code that calls
+`process.binding()` to import additional code or data. This leads us into the
+next segment...
+
+
+### C++ Source (`src/`)
+
+Node.js is built ontop of a C++ base for both bootstrapping and performance
+reasons. This also helps for interoperability with Node.js's underlying
+JavaScript engine, V8, which is also written in C++.
+
+Some of the C++ source files export JavaScript code defined in C++ via the V8
```

Comment:
If I’m not misunderstanding you (i.e. you are referring to e.g. the methods from node_file.cc), I’d leave out the `JavaScript` in `JavaScript code` or phrase it as `export methods/functions to JavaScript` or something like that